test/lib/sfge/SfgeDfaEngine.test.ts

```diff
@@ -1,45 +1,136 @@
 	}
 }
 
 describe('SfgeDfaEngine', () => {
-	describe('#processStdout()', () => {
-		it('When GraphEngine finds violations, they are converted into RuleResult objects', async () => {
+	describe('#parseViolations()', () => {
+		it('When output contains violations, they are converted into RuleResult objects', async () => {
 			// ==== SETUP ====
 			const testEngine = new TestableSfgeEngine();
 			await testEngine.init();
 			const spoofedOutput = fs.readFileSync(path.join('test', 'code-fixtures', 'sfge-results', 'one_violation.txt')).toString();
 
 			// ==== TESTED METHOD ====
-			const results = testEngine.processStdout(spoofedOutput);
+			const results = testEngine.parseViolations(spoofedOutput);
 
 			// ==== ASSERTIONS ====
 			expect(results.length).to.equal(1, 'Should be results');
 		});
 
-		it('When GraphEngine finds no violations, results are empty', async () => {
+		it('When output contains no violations, results are empty', async () => {
 			// ==== SETUP ====
 			const testEngine = new TestableSfgeEngine();
 			await testEngine.init();
 			const spoofedOutput = fs.readFileSync(path.join('test', 'code-fixtures', 'sfge-results', 'no_violations.txt')).toString();
 
 			// ==== TESTED METHOD ====
-			const results = testEngine.processStdout(spoofedOutput);
+			const results = testEngine.parseViolations(spoofedOutput);
 
 			// ==== ASSERTIONS ====
 			expect(results.length).to.equal(0, 'No results should have been returned');
 		});
 	});
+
+	describe('#parseError', () => {
+		it('Returns delineated error message if available', async () => {
+			// ==== SETUP ====
+			const testEngine = new TestableSfgeEngine();
+			await testEngine.init();
+			const spoofedOutput = fs.readFileSync(path.join('test', 'code-fixtures', 'sfge-results', 'delineated_error_without_partial_results.txt')).toString();
+			// ASSUMPTION: Line 21 of the spoofed output is where the delineation is.
+			const expectedError = spoofedOutput.split('\n').slice(21).join('\n');
+
+			// ==== TESTED METHOD ====
+			const errorMessage = testEngine.parseError(spoofedOutput);
+
+			// ==== ASSERTIONS ====
+			expect(errorMessage).to.equal(expectedError, 'Wrong error found');
+		});
+
+		it('Returns entire message when no delineated error message is available', async () => {
+			// ==== SETUP ====
+			const testEngine = new TestableSfgeEngine();
+			await testEngine.init();
+			const spoofedOutput = fs.readFileSync(path.join('test', 'code-fixtures', 'sfge-results', 'undelineated_error.txt')).toString();
+
+			// ==== TESTED METHOD ====
+			const errorMessage = testEngine.parseError(spoofedOutput);
+
+			// ==== ASSERTIONS ====
+			expect(errorMessage).to.equal(spoofedOutput, 'Wrong error found');
+		});
+	});
+
+	describe('#processExecutionFailure', () => {
+		beforeEach(() => {
+			Sinon.createSandbox();
+		})
+
+		afterEach(() => {
+			Sinon.restore();
+		});
+
+		it('If partial results are present, they are returned and errors are logged', async () => {
```

Comment:
This test case validates that the old functionality still applies where appropriate.